### PR TITLE
Automated delta selection

### PR DESCRIPTION
**Pull request by jovianyu**
Initial implementation of an automated delta selection process.

The new selection process occurs in `identifyConsensus.py`

For testing purposes, the new selection process can be used by setting the flag `-d` with `identifyConsensus.py`, with the caveat that the arguments for `-n` will expect directories for each HotNet2 run and not paths to the `results.json` files.

In `identifyConsensus.py`, `choose_results` takes the directories for each network, sorts group of `results.json` to find the delta that corresponds to the lowest p-value for a CC, followed by the smallest CC size with this p-value, and finally the lowest delta that has the CC size with the p-value above. This reduces the input directory to a single `results.json` file per network to be used for the rest of the consensus procedure.

### CODE REVIEW

File: identifyConsensus.py [hunk 1]
```diff
@@ -10,12 +10,45 @@ def get_parser():
                         nargs="*", help='Paths to HotNet(2) results files.', required=True)
     parser.add_argument('-n', '--networks', nargs="*", required=True,
                         help='List the network that corresponds to each results file.')
+    parser.add_argument('-d', '--delta_choice', action='store_true',
+                        help='Test automated delta selection. Expects a directory of results for each network.')
```
Comment (mdml) [inline identifyConsensus.py]: Why does this say "test"?

Comment (jovianyu) [inline identifyConsensus.py]: Ah, I should change that, had meant for the flag to be used differently, initially.

File: identifyConsensus.py [hunk 2]
```diff
@@ -10,12 +10,45 @@ def get_parser():
...
+            # Loading results json files and adding entries if k (size) > min_cc_size into a list of
+            # tuples: ( delta, k, p-value, results json file )
+            for results_file in networks_results[network]:
+                with open( results_file ) as f: obj = json.load( f )
+                result_delta_list = [ ( float( obj['parameters']['delta'] ), int( k ), stats['pval'], results_file )
+                                      for k, stats in obj['statistics'].iteritems()
+                                      if int( k ) >= min_cc_size ]
+                delta_list.extend( result_delta_list )
+            # Choosing results json file with the lowest p-value, then the lowest k, then the lowest delta
+            for i in [2, 1, 0]:
```
Comment (mdml) [inline identifyConsensus.py]: This is nice, but wouldn't it be easier to just create a list of (P-value, k, \delta) tuples, sort them, and then take the first entry?

For example:

    results_files.append( sorted(delta_list)[0][3] )

**Comment by mdml** [comment]
I haven't tested this yet, but otherwise looks good to me besides my one comment about simplifying the code.

Could you write some example commands for testing this implementation that use the example files in `example/`? You might need to create a second network or heat file to do the testing.

**Comment by jovianyu** [comment]
Currently, the example works up until the consensus step (manual entry of the config file contents as flags in bash does work, however):
```
python makeRequiredPPRFiles.py @example/configs/influence_matrix.config
python makeRequiredPPRFiles.py @example/configs/influence_matrix2.config
python runHotNet2.py @example/configs/simple.config
python runHotNet2.py @example/configs/simple2.config
```

Then, this step seems to not be recognizing the flags in the config file:
`python identifyConsensus.py @example/configs/consensus.config`

**Comment by mdml** [comment]
Changed the example consensus config file to have one argument per line, as suggested [in the docs](https://docs.python.org/2.7/library/argparse.html#fromfile-prefix-chars). @jovianyu: your example works for me now.

**Pull request merged**